app/gui/project-manager-shim-middleware/projectManagement.ts

```diff
@@ -60,6 +60,42 @@ export async function uploadBundle(
   return bumpMetadata(targetPath, directory, name ?? null)
 }
 
+/** Create a .tar.gz enso-project bundle. */
+export function createBundle(directory: string): Promise<Buffer> {
+  const readableStream = tar.c(
+    {
+      z: true,
+      C: directory,
+    },
+    ['.'],
+  )
+  return new Promise((resolve, reject) => {
+    const chunks: Buffer[] = []
+    readableStream.on('data', (data) => chunks.push(data))
+    readableStream.on('end', () => resolve(Buffer.concat(chunks)))
+    readableStream.on('error', reject)
+  })
+}
+
+/** Unpack a .tar.gz enso-project bundle into a temporary directory */
+export async function unpackBundle(bundle: stream.Readable): Promise<string> {
+  const ensoProjectsDirectory =
```

Comment:
is it correct that in shim we don't respect the root directory setting ?

Reply:
I updated the logic. Right now we create a `{projects-root}/cloud-{projectId}` directory for the cloud project. In the future, we will make it hidden. I suppose it requires using a native library for Windows.